src/app/api/staff/mark-story/[staff_pick_id]/route.ts

```diff
@@ -0,0 +1,99 @@
+import prisma from "@/lib/prisma";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { patchSchema } from "../schema";
+
+interface Params {
+  staff_pick_id: string;
+}
+export async function DELETE(req: NextRequest, { params }: { params: Params }) {
+  try {
+    const { staff_pick_id } = params;
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+
+    if (!user || !user.roles.includes("EDITOR")) {
+      return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    }
+    const staffPickOld = await prisma.staffPick.findUnique({
+      where: { id: staff_pick_id },
+    });
+    if (!staffPickOld) {
+      return NextResponse.json({ error: "Not found" }, { status: 404 });
```

Comment:
is it possible for this to occur when params.staff_pick_id is undefined? Is it possible to also check the type of staff_pick_id and return status 400 if it's not the correct type / format?

Reply:
It is impossible for the staff_pick_id to be undefined, the front_end have to type the exact url and the staff_pick_id is part of the url. I can use schema to check if staff_pick_id is a valid type which I will change it latter.

Reply:
sorry for the comment, then.  Thanks, somehow missed that